src/driver/postgres/PostgresQueryRunner.ts

```diff
@@ -4160,7 +4160,11 @@ export class PostgresQueryRunner
         const result: [{ version: string }] = await this.query(
             `SELECT version()`,
         )
-        return result[0].version.replace(/^PostgreSQL ([\d.]+) .*$/, "$1")
+
+        // Examples:
+        // Postgres: "PostgreSQL 14.10 on x86_64-pc-linux-gnu, compiled by gcc (GCC) 8.5.0 20210514 (Red Hat 8.5.0-20), 64-bit"
+        // Yugabyte: "PostgreSQL 11.2-YB-2.18.1.0-b0 on x86_64-pc-linux-gnu, compiled by clang version 15.0.3 (https://github.com/yugabyte/llvm-project.git 0b8d1183745fd3998d8beffeec8cbe99c1b20529), 64-bit"
+        return result[0].version.replace(/^PostgreSQL ([\d.]+).*$/, "$1")
```

Comment:
Ideally we'd get the Postgres version number from `server_version_num` instead of parsing the string.

> Software developers should use server_version_num (available since 8.2) or [PQserverVersion](https://www.postgresql.org/docs/current/libpq-status.html#LIBPQ-PQSERVERVERSION) instead of parsing the text version.

https://www.postgresql.org/docs/current/functions-info.html#FUNCTIONS-INFO-VERSION
https://www.postgresql.org/docs/current/runtime-config-preset.html#GUC-SERVER-VERSION-NUM

This has been available since Postgres 8.2 which is quite old now. Do we have a minimum supported Postgres version?

Reply:
`SHOW server_version_num` would output something like `170002` - which can be used to compare versions, but that would mean changing the rest of the implementation too, to use this format.

`SHOW server_version` would output something like this:

```
9.3.25
9.6.24
14.10
15.0
16.6 (Debian 16.6-1.pgdg120+1)
17.2 (Debian 17.2-1.pgdg120+1)
11.2-YB-2.8.8.0-b0
11.2-YB-2.18.1.0-b0
```

I could also use that instead of `SELECT version()`.

Reply:
Actually, I just noticed the reason why we are using `SELECT version()`: #9318

https://docs.aws.amazon.com/redshift/latest/dg/c_unsupported-postgresql-functions.html

Let's leave it like this to have broader compatibility.